src/metabase/driver/clickhouse.clj

```diff
@@ -32,7 +37,9 @@
                               :test/jvm-timezone-setting       false
                               :connection-impersonation        false
                               :schemas                         true
-                              :datetime-diff                   true}]
+                              :uploads                         true
+                              :datetime-diff                   true
+                              :upload-with-auto-pk             false}]
```

Comment:
Why is it disabled?

Reply:
Theoretically, we could add a UUID instead. Or, maybe better, a CSV line number.

Reply:
Maybe UUID is better long term, but that would be out of scope. If we went with line number, we'd need to be careful about appending, particularly as we're not protecting against them happening concurrently.

Reply:
Fair question. My reasoning for delaying adding auto-incrementing PKs for Clickhouse is based on:
(a) If we use UUIDs, we need to extend the possible types for auto-generated PKs from integers to UUIDs. That's probably fine, but it would be nice if they were all the same type across DBMS systems. Assuming Clickhouse might support an auto-incrementing type in the future ([proposal here](https://github.com/ClickHouse/ClickHouse/issues/56228#issuecomment-1789933636)), it might be better to wait for that rather than commit to UUIDs.
(b) We don't actually use these auto-generated PK columns in Metabase yet, so I think it's okay not to use them. The idea behind them is in the future we can use them to allow users to edit rows in the UI without worrying about PKs.